### ISSUE TITLE
Password-only authentication fails with coredis (async mode)

### ISSUE THREAD

**Issue by zoltan-fedor** [alisaifee/limits#98]
Hi,
I was testing out the async mode, so I can add that to the `Falcon-limiter` package and wanted to test with a password-protected Redis instance, but failed.

I went to check and saw that the `coredis` now supports authentication (including username / password with `v2.2.1` - thank you @alisaifee !!!), see https://coredis.readthedocs.io/en/stable/release_notes.html?highlight=password#v2-2-1

I know that the async support is still in experimental mode, but I am hoping that this could still be fixed.
Thanks!

Below is the code and the error (the issue is the same if the password is provided via the RATELIMIT_STORAGE_OPTIONS):
```
import asyncio
from limits import parse
from limits.storage import storage_from_string
from limits.aio.strategies import MovingWindowRateLimiter

redis = storage_from_string("async+redis://:MyPassword@localhost:6379") 

moving_window = MovingWindowRateLimiter(redis)
one_per_minute = parse("1/minute")

async def hit():
   return await moving_window.hit(one_per_minute, "test_namespace", "foo")

loop = asyncio.get_event_loop()
loop.run_until_complete(hit())
loop.close()
```

The error:
```
Traceback (most recent call last):
  File "/home/myuser/.local/share/virtualenvs/test-limits-kWUTuU1U/lib/python3.8/site-packages/coredis/connection.py", line 468, in connect
    await self._connect()
  File "/home/myuser/.local/share/virtualenvs/test-limits-kWUTuU1U/lib/python3.8/site-packages/coredis/connection.py", line 728, in _connect
    await self.on_connect()
  File "/home/myuser/.local/share/virtualenvs/test-limits-kWUTuU1U/lib/python3.8/site-packages/coredis/connection.py", line 497, in on_connect
    if nativestr(await self.read_response()) != "OK":
  File "/home/myuser/.local/share/virtualenvs/test-limits-kWUTuU1U/lib/python3.8/site-packages/coredis/connection.py", line 529, in read_response
    raise response
coredis.exceptions.ResponseError: wrong number of arguments for 'auth' command

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/myuser/.local/share/virtualenvs/test-limits-kWUTuU1U/lib/python3.8/site-packages/coredis/client.py", line 221, in execute_command
    await connection.send_command(*args)
  File "/home/myuser/.local/share/virtualenvs/test-limits-kWUTuU1U/lib/python3.8/site-packages/coredis/connection.py", line 564, in send_command
    await self.connect()
  File "/home/myuser/.local/share/virtualenvs/test-limits-kWUTuU1U/lib/python3.8/site-packages/coredis/connection.py", line 472, in connect
    raise ConnectionError()
coredis.exceptions.ConnectionError
```

EDIT: I just realized, that maybe I should have raised this against `coredis` instead?

**Comment by alisaifee** [comment]
Thanks for the report. Yes both aysnc mode and coredis itself are pretty experimental right now, but I'm pretty motivated to move them along quickly :) 
Yes the issue is probably in coredis, but I'll add a few more tests in limits to verify and correct it.

**Comment by zoltan-fedor** [comment]
Great thanks!!!

**Comment by alisaifee** [comment]
Fixed in `coredis >= 2.2.3`

**Comment by zoltan-fedor** [comment]
Thanks, I have tested it, it works great now with `v2.2.3`
